File: apps/api/src/core/supabase.py
"""
Supabase client initialization.
"""
from supabase import create_client, Client
from .config import settings

def get_supabase_client() -> Client:
    """
    Get initialized Supabase client.
    
    Returns:
        Client: Supabase client instance
    """
    return create_client(settings.supabase_url, settings.supabase_key)

def get_supabase_service_client() -> Client:
    """
    Get initialized Supabase client with service role key (admin).
    
    Returns:
        Client: Supabase admin client instance
    """
    if not settings.supabase_service_key:
        raise ValueError("SUPABASE_SERVICE_KEY not configured")
    return create_client(settings.supabase_url, settings.supabase_service_key)

File: apps/engine/src/db/supabase.ts
import { createClient, SupabaseClient } from '@supabase/supabase-js';
import { logger } from '../utils/logger.js';

let supabaseClient: SupabaseClient | null = null;

/**
 * Get Supabase client for Node.js Engine
 * Uses service role key for database access
 */
export function getSupabaseClient(): SupabaseClient {
    if (!supabaseClient) {
        const supabaseUrl = process.env.SUPABASE_URL;
        const supabaseKey = process.env.SUPABASE_SERVICE_KEY;

        if (!supabaseUrl || !supabaseKey) {
            throw new Error('Missing SUPABASE_URL or SUPABASE_SERVICE_KEY environment variables');
        }

        supabaseClient = createClient(supabaseUrl, supabaseKey, {
            auth: {
                autoRefreshToken: false,
                persistSession: false
            }
        });

        logger.info('Supabase client initialized');
    }

    return supabaseClient;
}
